universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -53,4 +55,14 @@ public QuestionDefinition update(QuestionDefinition definition) {
                     .map(question -> question.getQuestionDefinition())
                     .collect(ImmutableList.toImmutableList()));
   }
+
+  private boolean isValid(QuestionDefinition definition) {
```

Comment:
We probably also want to run this through a url encoder. I think any string where the encoded value isn't equal to the raw value should be denied. We can also validate that client side, but as a general rule client side validation should only be done as a service to the user, not to ensure things are legitimate on the server side.

Reply:
That definitely sounds something we want to do! We probably want to do that for all string fields. For now, let me add a super basic path pattern matching check. PTAL.

Reply:
Done.